incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -222,7 +223,14 @@ func (r *ConfigReconciler) syncObjectReconcilers(ctx context.Context, inst *api.
 // syncActiveReconcilers syncs object reconcilers for types that are in the Spec. If an object reconciler exists, it sets
 // its mode according to the Spec; otherwise, it creates the object reconciler.
 func (r *ConfigReconciler) syncActiveReconcilers(ctx context.Context, inst *api.HNCConfiguration) error {
+	// exist keeps track of existing types in the `config` singleton.
+	exist := gvkSet{}
 	for _, t := range inst.Spec.Types {
+		// If there are multiple configurations of the same type, we will follow the first
+		// configuration and ignore the rest.
+		if r.ensureNoDuplicateTypeConfigurations(inst, t, exist) {
```

Comment:
I think this would look cleaner if it was `!r.ensure()` instead of `r.ensure()`. That is, return false on "failure" so we skip. This makes it look like "we've successfully shown there's no conflict so we're going to do nothing" :)

Reply:
Thanks for the suggestion! Done.